test/functional/feature_settings.py

```diff
@@ -55,6 +55,13 @@ def run_test(self):
             fp.write("invalid json")
         node.assert_start_raises_init_error(expected_msg='Unable to parse settings file', match=ErrorMatch.PARTIAL_REGEX)
 
+        # Test invalid type
+        with settings.open("w") as fp:
+            json.dump({"blockversion": False}, fp)
+        node.assert_start_raises_init_error(
+            expected_msg='Error: Failed loading settings file:\n- Negating of -blockversion is meaningless and therefore forbidden',
```

Comment:
In commit "Disallow negation of blockversion" (fa1c4c9c7d10c11b0a2324ff0377919fa9fbc2fe)

I don't think having this second commit in this PR is a good idea. It's not directly related to the extra logging that this PR provides (which seems unambiguously good), and it's not a backwards compatible change, and it diverges from previous and still widespread behavior where -noarg is equivalent to arg=0. Changing command line processing could be good but I think it'd better to do it in a dedicated PR with more consistency and without tying it to another change

Reply:
ok, split commit into new pull